atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/internalschema/metrics/MetadataCoordinationServiceMetrics.java

```diff
@@ -129,10 +129,10 @@ private static void registerMetricForTransactionsSchemaVersionAtTimestamp(Metric
                         log,
                         Clock.defaultClock(),
                         metricName,
-                        () -> {
-                            ValueAndBound<TimestampPartitioningMap<Integer>> mapAndBound = timestampMapSupplier.get();
-                            return getVersionAtTimestamp(mapAndBound.value(), timestampQuery.apply(mapAndBound));
-                        }));
+                        () -> timestampMapSupplier.get()
+                                .map(mapAndBound -> getVersionAtTimestamp(
+                                        mapAndBound.value(), timestampQuery.apply(mapAndBound)))
+                                .orElse(null)));
```

Comment:
I was concerned about the null value here also, but seems like returning null here is safe.

Reply:
Yep. For metrics null basically means don't plot a data point / don't return a value